### PR TITLE
yuzu Progress Report August 2020

### PR DESCRIPTION
**Pull request by goldenx86**
Attack of the weebs

### CODE REVIEW

File: site/content/entry/yuzu-progress-report-sep-2020/index.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+## I have no motion, and I must shake
+
+If your favorite controller has no support for motion controls (such as Xbox controllers), don't worry, not everything is lost. After developing the code necessary to support this feature into the emulator, [german77](https://github.com/german77) also made this follow-up PR to [implement a button to simulate a shake](https://github.com/yuzu-emu/yuzu/pull/4677). This will allow you to configure any key of your liking to work as if you were physically swinging your controller and activating the in-game motion mechanics.
+
+While this should work with games such as `Super Mario Odyssey` and others, do note it won't be able to cover all cases or titles. Every game is different, and the required input is not always just a shake.
+
+## Remember the Rumble Pak?
+
+Also thanks to [german77](https://github.com/german77), yuzu can now proudly boast [Rumble support](https://github.com/yuzu-emu/yuzu/pull/4291). Any SDL2 compatible device will vibrate when asked by your Switch games. This includes but is not limited to DualShock, Xbox, and Nintendo controllers.
```
Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:47]: Add `:` after `limited to`

File: site/content/entry/yuzu-progress-report-sep-2020/index.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+HD Rumble is not possible, yet. The SDL2 specification is incompatible, and not everyone will play with Nintendo peripherals. This means that a Pro Controller or a pair of Joy-Cons cannot reach the same level of precision as the Switch. Additionally, some games seem to use different ways to ask for rumble and are not supported yet. We are investigating this.
+
+Soon, german77 has plans to add native Joy-Con pairing support. This will allow for real HD Rumble compatibility.
+
+## Video decoding, or how to suffer for so long with a single codec
+
+It's finally here! After such a long wait, and thanks to the work from [ogniK](https://github.com/ogniK5377) and [epicboy](https://github.com/ameerj), [yuzu can now play videos in-game!](https://github.com/yuzu-emu/yuzu/pull/4729) For now, `H.264` and `VP9` videos are supported, which thankfully seems to cover all released games until now. Support for `H.265` and `VP8` will come later.
```
Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:55]: Add `only` after `For now,`

File: site/content/entry/yuzu-progress-report-sep-2020/index.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    "./mk8.mp4| Video Killed the Radio Star (Mario Kart 8 Deluxe)"
+  >}}
+  
+This is possible by using [FFmpeg.](https://ffmpeg.org/) For now, decoding happens in a single CPU thread, and support for multiple threads and GPU decoding is currently being worked on.
+
+We have an entire article dedicated to showing how this feature came to be, how it works, its challenges, and current limitations in more detail [here.](https://yuzu-emu.org/entry/yuzu-nvdec-emulation/) Please take your time to read it: it also includes a list of games confirmed working and not-so-working yet.
+
+## An emulator inside an emulator
+
+`Super Mario 3D All-Stars` is a special case, in several aspects. For starters, the game is just a container for several other binary executables (known as `Program NCAs`), each one with their own Title ID. Step one to get this game to boot is to handle [a particular case like this](https://github.com/yuzu-emu/yuzu/pull/4675), a job [Morph](https://github.com/Morph1984) did.
```
Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:67]: Remove comma before `in several aspects.`

Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:67]: Change `to handle a particular case like this` to `to correctly handle these multi-content games`

File: site/content/entry/yuzu-progress-report-sep-2020/index.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  
+## Vulkan changes
+
+`VK_KHR_timeline_semaphore` is a relatively recent Vulkan extension only supported by *desktop or laptop* GPUs. In the past, resources were being assigned with `VkFence` and freed when arbitrarily asked, leading to a possible small pileup. With `VK_KHR_timeline_semaphore`, each resource is tied to a periodically checked "GPU tick," and you only need to verify on which tick you are at the moment to know if that resource is now free.
+
+{{< imgs
+    "./vk.png| Our implementation is not that complex. This is just an example provided by Khronos"
+  >}}
+
+[Adding support for timeline semaphores on yuzu](https://github.com/yuzu-emu/yuzu/pull/4674) helps simplify the code a lot, improves VRAM use a small amount, and paves the way for the finalization of the `Texture Cache rewrite`. Thank you, [Rodrigo](https://github.com/ReinUsesLisp), for its addition! The result should be transparent for the user with no perceptible change besides a bit of VRAM savings.
```
Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:87]: Change `semaphores on yuzu` to `semaphores in yuzu`

File: site/content/entry/yuzu-progress-report-sep-2020/index.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+## Bug fixes and improvements
+
+A missing feature in the input rewrite was the ability to merge two separate single left and right Joy-Cons into a single dual Joy-Con, a function `Super Mario Odyssey` needs. [Morph](https://github.com/Morph1984) [implemented `MergeSingleJoyasDualJoy`](https://github.com/yuzu-emu/yuzu/pull/4629), so this is no longer a problem for Mario. He can return to work once again for nothing. Thirty-nine years and no rewards for the best hero the Mushroom Kingdom ever saw.
+
+A small PSA for our Linux users, the [Qt required version has been raised to `5.12.8`](https://github.com/yuzu-emu/yuzu/pull/4638) now. Thanks [Morph](https://github.com/Morph1984) for the update! We need those new Plasma functions.
```
Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:93]: Change `Qt required version` to `required Qt version`

File: site/content/entry/yuzu-progress-report-sep-2020/index.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+Users have been reporting input latency, so [Tobi](https://github.com/FearlessTobi) decided to [test some changes](https://github.com/yuzu-emu/yuzu/pull/4643) and see if the situation improves. To further mitigate this issue if you experience it, be sure to enable `Radeon Anti-Lag` or `Nvidia Ultra-Low Latency` in your respective driver control panel. Nothing makes you ragequit more than losing in Smash due to lag.
+
+Now [Gamecube controllers will be autoconfigured!](https://github.com/yuzu-emu/yuzu/pull/4618) Thanks to work done by [german77](https://github.com/german77), now the Gamecube Adapter controllers will show up as devices in the input window, and their buttons will be mapped automatically. Laziness is the best.
```
Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:97]: Change `Now Gamecube controllers will be autoconfigured!` to `Gamecube controllers will now be autoconfigured!`

Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:97]: Remove `now the` before `Gamecube Adapter controllers`

File: site/content/entry/yuzu-progress-report-sep-2020/index.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+`Clubhouse Games: 51 Worldwide Classics` now manages to boot and, with a save, load the board games. [Morph](https://github.com/Morph1984) managed this by [implemention the `Get/SetNpadHandheldActivationMode` function](https://github.com/yuzu-emu/yuzu/pull/4683). The game will render almost everything in black, for now, so don't consider it playable yet.
```
Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:99]: Change `implemention` to `implementing`

Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:99]: Change `The game will render almost everything in black, for now, so don’t consider it playable yet.` to `However, the game will still render almost everything in black, so don't consider it playable yet.`

File: site/content/entry/yuzu-progress-report-sep-2020/index.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+{{< single-title-imgs
+    "Paint it all black (Clubhouse Games: 51 Worldwide Classics)"
+    "./51a.png"
+    "./51b.png"
+  >}}
+
+If you ran a game in fullscreen and a pop-up needed to show up (for example, naming a character), it used to render behind the main yuzu window, forcing you to exit fullscreen, focus the pop-up, address the reason for it, and then continue with your game. Quite annoying indeed. [Morph](https://github.com/Morph1984) is our savior for this one, by [using the `Qt::WindowStaysOnTopHint` flag](https://github.com/yuzu-emu/yuzu/pull/4728), the pop-ups will now show up on top of your game. Thanks a lot, poyo.
```
Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:107]: Change `show up` to `display`

Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:107]: Change `address the reason for it` to `complete the action`

File: site/content/entry/yuzu-progress-report-sep-2020/index.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+{{< single-title-imgs
+    "Just in time for Ampere, eh! Good luck getting one for now (Super Smash Bros. Ultimate)"
+    "./SSBU-broken.mp4"
+    "./SSBU-fixed.mp4"
+  >}}
+
+As mentioned in [July's progress report](https://yuzu-emu.org/entry/yuzu-progress-report-jul-2020/), [toastUnlimited](https://github.com/lat9nq) continued to work in solving crashes only experienced with recent Nvidia Vulkan drivers in Linux. Thanks to work done by DXVK, the [current implementation aims to be global](https://github.com/yuzu-emu/yuzu/pull/4724), and so far we haven't got any report on getting the issue or any sort of regression. Please feel free to contact us if you are still affected by this issue.
```
Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:119]: Change `continued to work in` to `continued to work on`

Comment (Schplee) [inline site/content/entry/yuzu-progress-report-sep-2020/index.md:119]: Change `the current implementation` to `Toast's current implementation`

**Review by Schplee — CHANGES_REQUESTED** [review]

File: site/content/entry/yuzu-progress-report-sep-2020/index.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
++++
+date = "2020-09-30T12:00:00-03:00"
+title = "Progress Report September 2020"
+author = "Honghoa"
+coauthor = "GoldenX86"
+forum = 310318
++++ 
+
+Greetings Yuz-ers! Welcome to September's progress report. This month we offer you actual dancing, pictures per second, more input fixes and additions, small Vulkan improvements, and the initial groundwork needed to get Super Mario 3D All-Stars playable.
```
[the remark this answers is not in the document]
Comment (goldenx86) [reply site/content/entry/yuzu-progress-report-sep-2020/index.md:9]: Zoomer...

File: site/content/entry/yuzu-progress-report-sep-2020/index.md (hunk 7 above)
Comment (Morph1984) [reply site/content/entry/yuzu-progress-report-sep-2020/index.md:99]: Also change `[Morph](https://github.com/Morph1984) managed this by [implemention the `Get/SetNpadHandheldActivationMode` function]` to `[Morph](https://github.com/Morph1984) managed this by [implementing the `Get/SetNpadHandheldActivationMode` functions]

**Review by Schplee — APPROVED** [review]

**Pull request merged**